akka-actor/src/main/scala/akka/actor/Timers.scala

```diff
@@ -37,8 +39,11 @@ trait Timers extends Actor {
     msg match {
       case timerMsg: TimerSchedulerImpl.TimerMsg ⇒
         _timers.interceptTimerMsg(timerMsg) match {
-          case OptionVal.Some(m) ⇒ super.aroundReceive(receive, m)
-          case OptionVal.None    ⇒ // discard
+          case OptionVal.Some(m) ⇒
+            // this is important for stash interaction, as stash will look directly at currentMessage #24557
+            actorCell.currentMessage = Envelope(m, sender())
+            super.aroundReceive(receive, m)
```

Comment:
I guess we could do an instanceof-check on `this` and only do the extra work/allocation when `Stash` is mixed in?

Reply:
I think that would be good